be/src/olap/null_predicate.h

```diff
@@ -92,7 +94,7 @@ class NullPredicate : public ColumnPredicate {
         }
     }
 
-    bool can_do_bloom_filter() const override { return _is_null; }
+    bool can_do_bloom_filter(bool ngram) const override { return ngram ? false : _is_null; }
```

Comment:
may be `bool can_do_bloom_filter(bool ngram) const override { return !ngram && _is_null; }` is better